life_expectancy/load_save_data.py

```diff
@@ -1,23 +1,77 @@
 """This model loads and saves the data"""
-from pathlib import Path
+#pylint: disable=R0903
 
+from pathlib import Path
+from typing import Any, Protocol
+import zipfile
 import pandas as pd
 
 
-def load_data(
-        input_file_path: Path
-    ) -> pd.DataFrame:
-    """
-    Load eu_life_expectancy_raw.tsv data from the data folder
-    Args:
-        input_file_path (Path): Path for the file to be loaded
-    Returns:
-        life_exp_raw_data (Pandas DataFrame): DataFrame with data to be cleaned
-    """
-    with open(input_file_path, "r", encoding="UTF-8") as file:
-        life_exp_raw_data = pd.read_csv(file, sep="\t")
+class FileReaderStrategy(Protocol):
+    """Reads the file"""
+
+    def __call__(self, file:Any) -> pd.DataFrame:
+        """
+        Reads the file and returns the data as a pandas DataFrame
+        Args:
+            file (Any): file object to read
+        Returns:
+            pd.DataFrame: The data read from the file as a pandas DataFrame
+        Raises:
+            NotImplementedError: This is a protocol method, so must be implemented by subclasses
+        """
+
+        raise NotImplementedError("This method should be implemented by subclasses")
```

Comment:
You don't need to define a NotImplementedError on a Protocol. In fact, because `__call__` already comes with any Python object, the only thing it would be checking is its signature.

Reply:
Thanks, I have removed NotImplementedError :)